core/services/offchainreporting2/delegate.go

```diff
@@ -75,82 +76,40 @@ func (Delegate) AfterJobCreated(spec job.Job)  {}
 func (Delegate) BeforeJobDeleted(spec job.Job) {}
 
 func (d Delegate) ServicesForSpec(jobSpec job.Job) (services []job.Service, err error) {
-	if jobSpec.Offchainreporting2OracleSpec == nil {
-		return nil, errors.Errorf("offchainreporting.Delegate expects an *job.Offchainreporting2OracleSpec to be present, got %v", jobSpec)
-	}
 	spec := jobSpec.Offchainreporting2OracleSpec
-
-	chain, err := d.chainSet.Get(jobSpec.Offchainreporting2OracleSpec.EVMChainID.ToInt())
-	if err != nil {
-		return nil, err
-	}
-
-	contract, err := offchain_aggregator_wrapper.NewOffchainAggregator(spec.ContractAddress.Address(), chain.Client())
-	if err != nil {
-		return nil, errors.Wrap(err, "could not instantiate NewOffchainAggregator")
-	}
-
-	contractFilterer, err := ocr2aggregator.NewOCR2AggregatorFilterer(spec.ContractAddress.Address(), chain.Client())
-	if err != nil {
-		return nil, errors.Wrap(err, "could not instantiate NewOffchainAggregatorFilterer")
+	if spec == nil {
+		return nil, errors.Errorf("offchainreporting.Delegate expects an *job.Offchainreporting2OracleSpec to be present, got %v", jobSpec)
 	}
 
-	contractCaller, err := ocr2aggregator.NewOCR2AggregatorCaller(spec.ContractAddress.Address(), chain.Client())
+	ocr2Provider, err := d.relayer.NewOCR2Provider(jobSpec.ExternalJobID, spec)
 	if err != nil {
-		return nil, errors.Wrap(err, "could not instantiate NewOffchainAggregatorCaller")
+		return nil, errors.Wrap(err, "error calling 'relayer.NewOCR2Provider'")
 	}
+	services = append(services, ocr2Provider)
 
-	ocrdb := NewDB(d.db.DB, spec.ID, d.lggr)
-
-	tracker := NewOCRContractTracker(
-		contract,
-		contractFilterer,
-		contractCaller,
-		chain.Client(),
-		chain.LogBroadcaster(),
-		jobSpec.ID,
-		d.lggr,
-		d.db,
-		ocrdb,
-		chain.Config(),
-		chain.HeadBroadcaster(),
-	)
-	services = append(services, tracker)
-
-	var peerID p2pkey.PeerID
-	if spec.P2PPeerID != nil {
-		peerID = *spec.P2PPeerID
-	} else {
-		peerID = chain.Config().P2PPeerID()
-	}
+	ocrDB := NewDB(d.db.DB, spec.ID, d.lggr)
 	peerWrapper := d.peerWrapper
 	if peerWrapper == nil {
 		return nil, errors.New("cannot setup OCR2 job service, libp2p peer was missing")
 	} else if !peerWrapper.IsStarted() {
 		return nil, errors.New("peerWrapper is not started. OCR2 jobs require a started and running peer. Did you forget to specify P2P_LISTEN_PORT?")
-	} else if peerWrapper.PeerID != peerID {
-		return nil, errors.Errorf("given peer with ID '%s' does not match OCR2 configured peer with ID: %s", peerWrapper.PeerID.String(), peerID.String())
-	}
-	bootstrapPeers, err := ocrcommon.GetValidatedBootstrapPeers(spec.P2PBootstrapPeers, chain)
-	if err != nil {
-		return nil, err
```

Comment:
Whats the reason behind moving this?

Reply:
chain is evm specific, all evm specific stuff is going inside the ethereum relay